app.js

```diff
@@ -6,95 +6,112 @@ var express = require('express');
+        if (!app.conf.port) {
+            app.conf.port = 8888;
         }
-        // import the route file
-        var route = require(__dirname + '/routes/' + fname);
-        route = route(app);
-        // check that the route exports the object we need
-        if(route.constructor !== Object || !route.path || !route.router) {
-            throw new Error('routes/' + fname + ' does not export the correct object!');
+        if (!app.conf.interface) {
+            app.conf.interface = '0.0.0.0';
         }
-        // all good, use that route
-        app.use(route.path, route.router);
-    }).then(function() {
-        // route loading is now complete, return the app object
-        return app;
+
+        // disable the X-Powered-By header
+        app.set('x-powered-by', false);
+        // disable the ETag header - users should provide them!
+        app.set('etag', false);
+        // enable compression
+        app.use(compression(app.conf.compressionLevel ? {level: app.conf.compressionLevel} : {}));
+        // use the JSON body parser
+        app.use(bodyParser.json());
+        // use the application/x-www-form-urlencoded parser
+        app.use(bodyParser.urlencoded({extended: true}));
+        // use the multipart/form-data
+        app.use(multer());
+        // serve static files from static/
+        app.use('/static', express.static(__dirname + '/static'));
+
+        resolve(app);
     });
+}
 
+/**
+ * Async load all routes for the app
+ * @param app
+ * @returns {bluebird}
+ */
+function loadRoutes (app) {
+    // get the list of files in routes/
+    return fs
+        .readdirAsync(__dirname + '/routes')
+        .map(function (fname) {
+            // ... and then load each route
+            // but only if it's a js file
+            if (!/\.js$/.test(fname)) {
+                return;
+            }
+            // import the route file
+            var route = require(__dirname + '/routes/' + fname);
+            route = route(app);
+            // check that the route exports the object we need
+            if (route.constructor !== Object || !route.path || !route.router) {
+                throw new Error('routes/' + fname + ' does not export the correct object!');
+            }
+            // all good, use that route
+            app.use(route.path, route.router);
+        }).then(function () {
+            // route loading is now complete, return the app object
+            return app;
+        });
 }
 
+/**
+ * Async create a web server
+ * @param app
+ * @returns {bluebird}
+ */
+function createServer(app) {
+    // return a promise which creates an HTTP server,
+    // attaches the app to it, and starts accepting
+    // incoming client requests
+    return new BBPromise(function (resolve) {
+        http.createServer(app).listen(
+            app.conf.port,
+            app.conf.interface,
+            resolve
+        );
+    }).then(function () {
+            app.logger.log('info',
```

Comment:
Indentation - there's one extra level which should be removed (same for the next two lines).